### PR TITLE
Revamp our doc structure.

### PR DESCRIPTION
**Pull request by troessner**
* This makes #422 prominent in our README and adds it to /docs
* Moves the "Working with rails" inside the README since this is something that trips up a lot of people.
* Makes sure every document in /docs (except for all the smells) is at least linked from the README so that people are aware of it (I assume almost nobody will scan /docs by default)
* Adds

>> See docs/.... for details.

to every smell, thus fixing #499

### CODE REVIEW

File: docs/versioning-policy.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Versioning Policy
```
Comment (mvz) [inline docs/versioning-policy.md]: Maybe call this file `Versioning-Policy.md` to match the other names in `docs/`?

File: docs/Non-ASCII-source-files.md [hunk 2]
```diff
@@ -1,9 +1,8 @@
 # Non ASCII source files
 
-
 `reek` doesn't offer a direct way to specify that your source file is encoded as UTF-8 etc, but there is a workaround if you need it:
 
-`reek` includes the class [Reek::RakeTask](http://reek.rubyforge.org/rdoc/classes/Reek/RakeTask.html), which makes it easy to run `reek` from a rakefile. And this task has an attribute `ruby_opts`, which is an array of strings to be passed as arguments directly to the Ruby interpreter. So you can use `reek` to check for smells in non-ASCII files by adding something like this to your rakefile: 
+`reek` includes the class [Reek::RakeTask](http://reek.rubyforge.org/rdoc/classes/Reek/RakeTask.html), which makes it easy to run `reek` from a rakefile. And this task has an attribute `ruby_opts`, which is an array of strings to be passed as arguments directly to the Ruby interpreter. So you can use `reek` to check for smells in non-ASCII files by adding something like this to your rakefile:
```
Comment (mvz) [inline docs/Non-ASCII-source-files.md]: I wonder if this tip is still relevant, since ruby assumes utf8 by default now AFAIK.

Comment (andyw8) [inline docs/Non-ASCII-source-files.md]: If this is kept, 'rakefile' should be 'Rakefile'.

File: docs/versioning-policy.md (hunk 1 above)
Comment (troessner) [inline docs/versioning-policy.md]: Agreed.:)

File: docs/Non-ASCII-source-files.md (hunk 2 above)
Comment (troessner) [inline docs/Non-ASCII-source-files.md]: Good point, I removed the file.

**Comment by troessner** [comment]
Removed Non-ASCII-source-files and renamed versioning-policy.md.

**Comment by mvz** [comment]
Good to go. Merging.

**Pull request merged**